<?xml version="1.0"?>
<launch>
  <arg name="LOG_LEVEL" default="Info"/>
  <arg name="debug" default="false"/>

  <arg unless="$(arg debug)" name="LAUNCH_PREFIX" value=""/>
  <arg     if="$(arg debug)" name="LAUNCH_PREFIX" value="gdb -ex run --args "/>

  <!-- Orocos target (gnulinux, xenomai, etc) -->
<!--  <arg name="OROCOS_TARGET" default="$(optenv OROCOS_TARGET)"/>-->
  <!-- Path to look for dynamically-loaded plugins and components (something like /lib/orocos) -->
<!--  <arg name="RTT_COMPONENT_PATH" default="$(env RTT_COMPONENT_PATH)"/>-->

  <node
    launch-prefix="$(arg LAUNCH_PREFIX)"
    name="velma_core_ve_body"
    pkg="subsystem_deployer" type="deployer"
    args="--DeployerName velma_core_ve_body -l $(arg LOG_LEVEL) -x $(find velma_core_ve_body)/config/velma_core_ve_body.xml -m velma_core_ve_body --" output="screen">
<!--    <env name="OROCOS_TARGET" value="$(arg OROCOS_TARGET)"/>
    <env name="RTT_COMPONENT_PATH" value="$(arg RTT_COMPONENT_PATH)"/>-->
<!--    <env name="LD_PRELOAD" value="libmalloc_hook.so" />-->
  </node>

  <group ns="velma_core_ve_body">
    <rosparam>
      safe:
        l_arm_damping_factors: [10, 10, 5, 5, 2, 1, 1]
        r_arm_damping_factors: [10, 10, 5, 5, 2, 1, 1]
        torso_damping_factor: 20
        arm_q_limits_lo: [-2.96, -2.09, -2.96, -2.09, -2.96, -2.09, -2.96]
        arm_q_limits_hi: [2.96, 2.09, 2.96, 2.09, 2.96, 2.09, 2.96]
        arm_dq_limits: [2, 2, 2, 2, 2, 2, 2]
        arm_t_limits: [100.0, 100.0, 100.0, 100.0, 100.0, 60.0, 60.0]
      TorsoPanTransmision:
        motor_offset: -88524
        joint_offset: 0.0
        gear: 158.0
        encoder_res: 20000.0
      TorsoPanTransmisionInv:
        motor_offset: -88524
        joint_offset: 0.0
        gear: 158.0
        encoder_res: 20000.0
        motor_constant: 0.00105
      HeadPanTransmision:
        motor_offset: 0.0
        joint_offset: 0.0
        gear: -100.0
        encoder_res: 8000.0
      HeadPanTransmisionInv:
        motor_offset: 0.0
        joint_offset: 0.0
        gear: -100.0
        encoder_res: 8000.0
        motor_constant: 0.0
      HeadTiltTransmision:
        motor_offset: 0.0
        joint_offset: 0.0
        gear: 100.0
        encoder_res: 8000.0
      HeadTiltTransmisionInv:
        motor_offset: 0.0
        joint_offset: 0.0
        gear: 100.0
        encoder_res: 8000.0
        motor_constant: 0.0
    </rosparam>
  </group>

</launch>
